src/libcore/ptr.rs

```diff
@@ -2866,7 +2866,7 @@ impl<T: ?Sized> NonNull<T> {
 
     /// Acquires the underlying `*mut` pointer.
     #[stable(feature = "nonnull", since = "1.25.0")]
-    pub fn as_ptr(self) -> *mut T {
+    pub const fn as_ptr(self) -> *mut T {
```

Comment:
can you add a `#[rustc_const_unstable(feature="non_null_as_ptr")]` to the method so we can stabilize its const evaluability explicitly and have a testing phase on nightly instead of making it insta-stable?

Reply:
Done